lib/gui/app.js

```diff
@@ -219,3 +220,47 @@ app.controller('HeaderController', function(SelectionStateModel, OSOpenExternalS
   };
 
 });
+
+app.controller('ApplicationController', function($state,
+                                          SupportedFormatsModel,
+                                          OSDialogService,
+                                          AnalyticsService,
+                                          SelectionStateModel) {
+
+  this.formats = SupportedFormatsModel;
+
+  /**
+   * @summary Select image
+   * @function
+   * @public
+   *
+   * @param {Object} image - image
+   *
+   * @example
+   * OSDialogService.selectImage()
+   *   .then(ApplicationController.dropImage);
+   */
+  this.dropImage = (image) => {
+    if (!SupportedFormatsModel.isSupportedImage(image.path)) {
+      OSDialogService.showError('Invalid dropped image', `${image.path} is not a supported image type.`);
+      AnalyticsService.logEvent('Invalid dropped image', image);
+      return;
+    }
+
+    SelectionStateModel.setImage(image);
+
+    AnalyticsService.logEvent('Dropped image', image);
+  };
+
+  /**
+   * @summary Check if the current state page matches the argument.
+   * @function
+   * @public
+   *
+   * @param String state - state
+   *
+   * @example
+   * if (ApplicationController.isStatePage('main')) doSomething();
+   */
+  this.isStatePage = $state.is;
```

Comment:
Can we rename this to `isState()`, for consistency with the original `$state.is` function?